src/apps/AppCenter/InstalledApps/InstalledApps.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Viewport } from '@aragon/ui'
+import { AppCenterAppType } from '../../../prop-types'
+import { ZoomableCards } from '../../../components/ZoomableCards'
+
+import AppsGrid from './AppsGrid'
+import AppContent from './AppContent'
+import AppVersions from './AppVersions'
+
+class InstalledApps extends React.Component {
+  static propTypes = {
+    apps: PropTypes.arrayOf(AppCenterAppType).isRequired,
+    openedAppName: PropTypes.oneOfType(
+      [PropTypes.string],
+      PropTypes.oneOf([null])
+    ),
+    onOpenApp: PropTypes.func.isRequired,
+    onRequestUpgrade: PropTypes.func.isRequired,
+  }
+  getAppByAppName(appName) {
+    return this.props.apps.find(app => app.appName === appName)
+  }
+  render() {
+    const { apps, openedAppName, onOpenApp } = this.props
+    return (
+      <Viewport>
+        {({ above, below }) => (
+          <ZoomableCards
+            currentId={openedAppName}
+            renderCards={({ card }) => (
+              <AppsGrid apps={apps} card={card} onOpenApp={onOpenApp} />
+            )}
+            renderOpenedCard={({ currentId }) => {
+              const app = this.getAppByAppName(currentId)
```

Comment:
> Technically we already have all the information about the opened app package in AppCenter, since that's where the associated state is held, and we could just pass that down to InstalledApps.

Yes, it’s always the same issue: we have the information during the opening transition and while it’s being opened, but we need to save it (here using the `currentId`) so that it can be displayed during the closing transition.

I think there must be a way to address this problem like react-spring does, by saving the rendered component itself before starting the closing transition. I need to look more into this, as it would be super useful for `SidePanel` too.

Reply:
Yeah, ran into exactly this while trying to make this bit leaner. And had to fight against this with the `SidePanel` too today ☹️ .